lib/inspec/profile.rb

```diff
@@ -72,9 +35,14 @@ def self.for_path(path, opts)
       new(reader, opts)
     end
 
+    def self.for_fetcher(fetcher, opts)
+      path, writable = fetcher.fetch
+      for_path(path, opts.merge(target: fetcher.target, writable: writable))
```

Comment:
wouldn't it be better to attach the fetcher to the profile and just read the target then from the fetcher?